tests/test_io.py

```diff
@@ -2355,3 +2357,79 @@ def test_basic_variable_get_set_result_correctness(fmu_path, result_handling):
+
+def _convert_char_array(names: list[str]) -> np.ndarray:
+    """Convert list of strings into MATLAB-style char array."""
+    max_len = max(map(len, names))
+    arr = np.full((max_len, len(names)), " ", dtype="U1")
+    for j, name in enumerate(names):
+        arr[:len(name), j] = list(name)
+    return arr
+
+def _save_mat_file(path: Path, data_info: np.ndarray, data_dict: dict, variable_names: list[str]) -> str:
+    """Helper to save a .mat file with given data and variable names."""
+    path = str(path)
+    mat_dict = {"dataInfo": data_info, "name": _convert_char_array(variable_names), **data_dict}
+    savemat(path, mat_dict, format="4")
+    return path
+
+@pytest.fixture(name="mat_file")
+def mat_file_fixture(tmp_path):
+    path = tmp_path / "mock_result_data.mat"
+
+    variable_names = ["time", "spring.phi_nominal", "torque.flange.phi", "@Diagnostics.step_time"]
+    data_info = np.array([[3, 2, 2, 3], [1, 2, 3, 2]], dtype=np.int32)
+
+    time_data_2 = np.arange(3, dtype=float)
+    data_2 = np.vstack([time_data_2, [1, 2, 3], [1, 2, 3], [1, 2, 3]])
+
+    time_data_3 = np.arange(1, 4, dtype=float)
+    data_3 = np.vstack([time_data_3, [1, 2, 3], [1, 2, 3]])
+
+    return _save_mat_file(path, data_info, {"data_2": data_2, "data_3": data_3}, variable_names)
+
+class TestResultHandlerForCalculatedDiagnostics:
+    def test_get_all_variable_names(self, mat_file):
+        result = ResultHandlerCalcDiags(mat_file)
+        assert len(result.get_variable_names()) == 4
+
+    def test_get_trajectories_from_all_matrices(self, mat_file):
+        result = ResultHandlerCalcDiags(mat_file)
+        for var in ["spring.phi_nominal", "torque.flange.phi", "@Diagnostics.step_time"]:
+            assert result.get_trajectory(var) is not None
+
+    def test_with_and_without_diagnostic_variables(self, tmp_path, mat_file):
+        result = ResultHandlerCalcDiags(mat_file)
+        assert result.get_trajectory("@Diagnostics.step_time") is not None
+
+        path = tmp_path / "mock_result_data_no_diag.mat"
+        variable_names = ["time", "spring.phi_nominal", "torque.flange.phi"]
+        data_info = np.zeros((2, len(variable_names)), dtype=np.int32)
+        _save_mat_file(path, data_info, {"data_2": np.zeros((4, 3))}, variable_names)
+
+        result_no_diag = ResultHandlerCalcDiags(str(path))
+        assert "@Diagnostics.step_time" not in result_no_diag.get_variable_names()
+
+    def test_edge_case_single_solver_point(self, tmp_path):
+        path = tmp_path / "mock_result_data_single_point.mat"
+        variable_names = ["time", "spring.phi_nominal"]
+        data_info = np.array([[2, 2], [1, 2]], dtype=np.int32)
+
+        time_data = np.array([0], dtype=float)
+        data_2 = np.vstack([time_data, [42]])
+
+        _save_mat_file(path, data_info, {"data_2": data_2}, variable_names)
+
+        result = ResultHandlerCalcDiags(str(path))
+        traj = result.get_trajectory("spring.phi_nominal")
+        assert traj.x == 42
```

Comment:
This looks a bit strange, it should not be a numpy array? Also, do we want a test where we assert trajectory values for other cases? No diagnostics, interpolation?

Reply:
Ah, I guess this works because `assert np.array([[42]]) == 42` seem to work for example. Not because `x` is a `float`.